### ISSUE TITLE
Make Mysql to save/load everything immediately

### ISSUE THREAD

**Issue by Phreag** [Siguza/VirtualPack#35]
Hey,
could you change your mysql db to save/load everything just when needed?
This would make it possible to use this plugin on a bungeecord network.

**Comment by Siguza** [comment]
This would require a major refactor of the entire data handling in VirtualPack.

The major problem is that, because of certain features, there are a whole lot of checks here and there, which require (at least parts of) the user data to be loaded at all times. Because of the horrible data format I chose when designing VirtualPack 2, loading a pack can take up to multiple seconds, and if UUID stuff has to be looked up, even up to half a minute. Such times are not even close to bearable when they should occur in response to a user action.
Another thing is that currently when saving, all entries in the database are dropped and the whole local stash is then inserted again. [I tried to fix this once](https://github.com/Siguza/VirtualPack/blob/master/src/main/java/net/drgnome/virtualpack/thread/VThreadSave.java#L52-L88) (look at the commented out code), but soon realised that this would just leave packs in the database, which had been deleted.
In addition, furnaces and brewing stands would no longer work, because they would have to be unloaded in order to be saved (or the data format would have to change).
And on top of that, there would have to be some locking mechanism implemented, or item losses/duplications and data corruption are bound to happen sooner or later.

TL;DR: Too much work, I'm sorry. I know this has been requested multiple times, and this is certainly on the list for VirtualPack 3 (although differently, and not so ill-designed), but VirtualPack 3 is not going to arrive anytime soon, because I just don't have the time to program at the moment.

**Comment by Phreag** [comment]
Yeah, the commented out code does not work, because it only updates the data.
It would be easy for it to get it working by using the SQL statement
"Delete From Table Where not in Data" (Something like this) this will delete all packs from the mysql which have been deleted.
And if you trigger save by PlayerQuitEvent and load by PlayerJoinevent it can be easy without big changes.

**Comment by Siguza** [comment]
Only that the table doesn't have any key, but yes, it would be possible with some work.
And "triggering save" done that easily either, because currently, everything is saved at once. Possible, but again, only with some work.
And the locking mechanism would still be needed.

The problem is just that I have a complete rewrite pending, and I have very little free time already, so if I'm gonna work on VPack at all, where should I spend my time? I don't even know if I'll get to that rewrite, like, ever. :/
